app/controllers/supplejack_api/harvester/records_controller.rb

```diff
@@ -116,6 +117,17 @@ def search_options_params
       def search_params
         params.require(:search).permit(['record_id', 'fragments.source_id', 'fragments.job_id'])
       end
+
+      def hints
+        indexes = SupplejackApi.config.record_class.collection.indexes.as_json.map { |index| index['key'].keys }.flatten
+        hints_hash = {}
+        search_params.each do |search_key, _v|
```

Comment:
```search_params.each_with_object({}) do |search_key, obj | ...```

would mean that you don't have to define the hints hash and would return the result of the loop.

Reply:
`hints` is a new method. You can write a text for it if you can mock `config` & `search+params`.

Reply:
I tried stubbing it, and things got really really messy.  I'll have another go.

Reply:
@DanHenton @EdwinRozario I've made both changes, thanks for the feedback.